### ISSUE TITLE
[enhancement]: Logging basic operations such as guard mode change on the UI without debug mode

### ISSUE THREAD

**Issue by dan1916** [homebridge-eufy-security/plugin#39]
### Describe the solution you'd like

Hello,

I was wondering if it is possible to see at-least some basic logs written to the Homebridge logs whenever this plugin is trying to perform some basic action (e.g. guard mode change). I see the debug mode option but that prints out way too much information.

Something along the lines of:
Eufy cam 1 Guard mode change requested : from HOME to AWAY
Eufy cam 1 Guard mode change request completed/errored out.

would be helpful.

**Comment by thieren** [comment]
Added this in #45

**Comment by dan1916** [comment]
@thieren thanks a lot. That was quick. 

Just a side question: Sometimes I have noticed that the set guard mode operation silently fails (e.g. I don't see the guard mode actually getting changed when I cross check it in the Eufy app, even after multiple refreshes of the Eufy app screen). Will this log help in understanding when a request to change the guard mode was submitted but later did not complete successfully? And can the plugin then retry this one more time? 

Changing guard mode is the number one functionality I use to automatically arm/disarm my Eufy cameras, and since the Eufy app severely lacks the functionality where one can do so automatically, I solely rely on this plugin to do it. It works great when it does. It's those 5% of the cases where despite toggling the guard mode from the Home app, it keeps on waiting on "Arming...." forever. One way to overcome is to reboot my Homebridge instance, at which point, it executes all such pending requests and then correctly finishes the execution.

**Comment by njuart** [comment]
> @thieren thanks a lot. That was quick. 
> 
> 
> 
> Just a side question: Sometimes I have noticed that the set guard mode operation silently fails (e.g. I don't see the guard mode actually getting changed when I cross check it in the Eufy app, even after multiple refreshes of the Eufy app screen). Will this log help in understanding when a request to change the guard mode was submitted but later did not complete successfully? And can the plugin then retry this one more time? 
> 
> 
> 
> Changing guard mode is the number one functionality I use to automatically arm/disarm my Eufy cameras, and since the Eufy app severely lacks the functionality where one can do so automatically, I solely rely on this plugin to do it. It works great when it does. It's those 5% of the cases where despite toggling the guard mode from the Home app, it keeps on waiting on "Arming...." forever. One way to overcome is to reboot my Homebridge instance, at which point, it executes all such pending requests and then correctly finishes the execution.

Reported something similar also here https://github.com/homebridge-eufy-security/plugin/issues/38

**Comment by dan1916** [comment]
My wild guess is the solocam2 that I have tend to go to sleep to preserve the battery life after sometime. The app seems to be able to do something to wake them up before changing the state. Homebase2 could be doing something similar. Just a guess.
Anyways, even if it fails, if it can have a functionality to retry X times, it would be super helpful. Because when these things are a part of an automation, expectation is that they always work as intended.

**Comment by thieren** [comment]
> Will this log help in understanding when a request to change the guard mode was submitted but later did not complete successfully?

The current PR only adds the log output. Since there was already one if you enabled debug there is not much difference than before. But to me it seems reasonable to add this info to the regular log - but that's only my view.
The maintainers could well argue otherwise and would also have a good argument, I think.

Also there is already an error log output if an error was detected. However I cannot say if in your case there is even an error thrown. So maybe the plugin never detects that something went wrong.

Anyway, depending on what exactly is causing your issue this could be out of scope of the plugin and would maybe have to updated in the eufy-security-client dependency.

> And can the plugin then retry this one more time?

This too has to be looked into. Also it has to be checked first if a retry will do the trick, since you and @njuart both state that if this fails that the station is stuck in "Arming...". I'm not quite sure if a seconds request will go through if this is the case.

My problem would be that I don't have the possibility to test this because I don't have a solocam2. But we'll can try to figure something out.

Just to clarify: Do you own both Homebase and Solocam2 and only the Solocam2 has this problem? Maybe I'm missing the point somewhere.

@njuart in #38 you mention that you have an indoor cam and the homebase. Does this error occur every time? And so does the changing state hang indefinitely or just some time (maybe 20 seconds)?

**Comment by njuart** [comment]
Yes it occurs every time, but on earlier betas it was ok. When I arm it doesn't show it's armed at all, after I open Eufy app and refresh, it becomes armed. And when I disarm it hangs in arming state. Honestly I'm not sure if it hangs indefinitely, maybe a couple of times it was disarmed by itself, maybe after few minutes or so.

**Comment by thieren** [comment]
I suggest to treat these as two separate issues, since they seem to have different causes despite having similar effects. The causes seem to be different since it happens all the time with @njuart and only sometimes with @dan1916

**Comment by lenoxys** [comment]
hello PR has been merge into latest beta

**Comment by dan1916** [comment]
I checked in the beta. I reports something like:

<camera_name> Request to change station guard mode to: Away.

It does not show whether the request was just submitted or executed successfully

**Comment by thieren** [comment]
I've added a log message that should be triggered if the mode change isn't done.

It's part of 2.0.1-beta.2. Until it's published you could download it directly from the repo.

Then there will be exactly one retry. But as stated before and from the information given I don't believe that the retry will do any good.

Please report if this helps.

**Comment by thieren** [comment]
Closed

not in the scope of this plugin.

See #38 and https://github.com/bropat/eufy-security-client/issues/161 for reference
